src/sql/finalize.sql

```diff
@@ -78,6 +78,11 @@ CREATE TYPE vector_index_stat AS (
     idx_options TEXT
 );
 
+CREATE TYPE vector_accum_state AS (
+	count BIGINT,
+	sum double precision[]
+);
```

Comment:
I feel uneasy about using a composite type as an aggregate state. `set_by_name` calls PostgreSQL functions so there are `sigsetjmp` cost. I'm not sure if it matters.

Reply:
I also think that it is a better way to ceate a new type and implement `IntoDatum` and `From Datum`. The cost will be less this way.